### PR TITLE
Implementation for Raw Artifacts with specified size

### LINKED ISSUE #1540: Implement support for generating test artifacts with a specified size for Raw
# Task Description

We need to be able to create test artifacts with a specified size.

# Classes of Interest

* `org.carlspring.strongbox.artifact.generator.RawArtifactGenerator`
* `org.carlspring.strongbox.testing.artifact.RawTestArtifact`

# Tasks

The following tasks will need to be carried out:
* [x] Apply the necessary changes to the `org.carlspring.strongbox.artifact.generator.RawArtifactGenerator`.
* [x] Apply the necessary changes to the `org.carlspring.strongbox.testing.artifact.RawTestArtifact`.
* [ ] Add tests for this.

# Task Relationships

This task:
* Is a sub-task of: #1534 
* Relates to: #1535 , #1536 , #1537, #1538 , #1582 
# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @fuss86
  * @sbespalov
  * @steve-todorov

### PR DESCRIPTION
**Pull request by anki2189**
# Pull Request Description

This pull request closes #1540 

# Acceptance Test

* [x] Building the code with `mvn clean install -Dintegration.tests` still works.
* [x] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [x] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [x] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Questions

* Does this pull request break backward compatibility? 
  * [ ] Yes
  * [x] No

* Does this pull request require other pull requests to be merged first? 
  * [ ] Yes, please see #...
  * [x] No

* Does this require an update of the documentation?
  * [ ] Yes, please see strongbox/strongbox-docs#{PR_NUMBER}
  * [x] No

### CODE REVIEW

File: strongbox-storage/strongbox-storage-api/src/test/java/org/carlspring/strongbox/storage/repository/RepositoryManagementTest.java [hunk 1]
```diff
@@ -108,7 +108,7 @@ private void artifactShouldBeCorrectlyResolvedAndUnique(Path artifact)
         throws IOException
     {
         assertThat(Files.exists(artifact)).isTrue();
-        assertThat(Files.size(artifact)).isEqualTo(1024L);
+        assertThat(Files.size(artifact)).isGreaterThan(1024L);
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-api/src/test/java/org/carlspring/strongbox/storage/repository/RepositoryManagementTest.java]: Just for my better understanding, why is this change required?

Comment (anki2189) [reply strongbox-storage/strongbox-storage-api/src/test/java/org/carlspring/strongbox/storage/repository/RepositoryManagementTest.java]: Because when file is generated using `TestFileUtils.generateFile` extra bytes are written to output stream hence size of artifact increases.

Comment (carlspring) [reply strongbox-storage/strongbox-storage-api/src/test/java/org/carlspring/strongbox/storage/repository/RepositoryManagementTest.java]: Right, thanks!

**Comment by carlspring** [comment]
Thank you very much for your help, @anki2189 ! :)
